application/txhandler.go

```diff
@@ -136,8 +137,10 @@ func (tm *txHandler) GetTxsForProposal(txn *badger.Txn, chainID uint32, height u
 			return nil, nil, err
 		}
 		if err := tm.PendingTxAdd(txn, chainID, height, []*objs.Tx{tx}); err != nil {
-			utils.DebugTrace(tm.logger, err)
-			return nil, nil, err
+			if !strings.Contains(err.Error(), "duplicate") {
```

Comment:
In an ideal world there would be some kind of `ErrDuplicate` that we could use `errors.Is` on, as there's a risk of being able to inject the string "duplicate" into an error somewhere. In this case, I think we're fine, but just something to keep in mind for later development.

Reply:
Totally agree!